### ISSUE TITLE
Feature: Allow for Publications to be disabled

### ISSUE THREAD

**Issue by colema18** [vasiliyfomichev/Slack-for-Sitecore#5]
add a enabled (or disabled) checkbox  to turn off a publication

**Comment by colema18** [comment]
merged into dev and ready for review/testing, then we can close.
- [ ] @vasiliyfomichev reviewed / tested
- [x] @vlukash reviewed / tested

**Comment by vasiliyfomichev** [comment]
After some thought, wouldn't the presence and absence of a publication definition item be an indicator of it being enabled or disabled? The way I look at it is it's not like we have a set number of publications that users can enable and disable, it's rather need-based, where users create what they need, so if they don't need it, they just won't create it. These types of settings do come in handy when users frequently enable/disable things, but I don't see them doing it with publications. It feel like adding this option is unnecessary. What do you guys think?

**Comment by vlukash** [comment]
As we do not tied to those publication items from our code we really can control publications by adding/deleting. But not sure if this behavior will be easier for end-user.
From my side I found "disabled" setting pretty useful while testing/fixing code. It's nice to have an ability to disable publication for some time to avoid spamming into a channel while configuring stuff. My only one concern is that we probably need to move "disabled" setting a little bit higher so it'll be visible without scrolling

**Comment by vasiliyfomichev** [comment]
it's handy for a developer like one of us, but in the field there isn't much tuning, the config is pretty straight forward, where you set it once the way you want it and perhaps modify it later. Another issue i see with this is usability, once you open the list of publications it is would be unclear which ones are enabled and which ones are disabled, unless we add custom styling, so users would have to click through all of them to figure it out. Also i've seen where users get lazy with on/off checkboxes, and end up never deleting anything, thinking, they might use it some time later, which they usually don't, so they shoot themselves in the foot and end up with a hard to manage mess :/ not that the publications have a high chance of getting out of control, but still, why enable it ))

**Comment by vasiliyfomichev** [comment]
this got me thinking, how about adding some usage analytics to the module? :) Google Analytics for instance

**Comment by colema18** [comment]
The disabled originated because of the developer testing side of things but I do think it would be useful in the real world. The two scenarios I think where it would be beneficial is that there potentially will be new events / triggers that are created that might not want to be enabled 100% of the time - especially when we start to have triggers tied to marketing / campaign events. The other instance is users that have multiple environments. At MEAU we have production (2 CD, 1 CM), staging, 2 QAs (development branch and feature branch), 3 CI Environments and then 4 developer machines. We would only want the Slack messages to occur on production (usually) - but we have scripts and tasks that restore the latest prod databases to the other environments when we are doing a build. When we are doing QA or Performance testing, we would want those publications disabled rather than deleted and then readded back in.

Ofcourse, if you feel strongly on the matter, we could get rid of the checkbox and AvailablePublications() would just return publications that have the Context set so to disable one would just have to clear that field.

GA is a nice idea. I am wondering if GA capabilities gets wrapped into the module or if GA Capabilities leverages the existing module but the publication target would just now be GA instead of Slack?

**Comment by vasiliyfomichev** [comment]
@colema18, good point! I'm onboard with the checkbox, and let's also add some styling associated with disabled items, perhaps make the item display name text color grey. Sound good?
